### ISSUE TITLE
Confusing usage of transformations

### ISSUE THREAD

**Issue by hiker** [stfc/PSyclone#978]
It is quite annoying (imho even for a more experienced dev) to keep track of which transformation to use: ``profile_transform`` comes from ``psyir/transformation`` (independent of API), while ``kernel extraction`` comes from ``domain/API/transformations`` (API dependent).

I know that ``TransInfo`` kept track of transformation in the past (and which afaik is broken atm since we started to move transforms to ``psyir/transformations``), but it still needs the user to know if a generic transformation or an API-specific transformation is to be used.

My suggestion: update this global object that stores transformation to handle API-specific transformation ... something like (I never liked that it came from psygen):
```
from psyclone.core.trans_nfo import TransInfo

all_trans = TransInfo()                # (1)
extract = all_trans.get("extract")     # would pick API-specific transform
profile = all_trans.get("profile")     # choses the generic API
```
In (1) we could supply the API as optional argument (but except maybe in tests the config object knows which API the user is working with).

**Comment by rupertford** [comment]
I agree that choosing the most appropriate transformation is not obvious and like your idea of a higher level that chooses the most appropriate. I must admit I'd given up to transinfo as I just import transformations explicitly and currently rely on the documentation to explain what to use (or more typically previous examples).

I just have a point about the location of the generic (non-api-specific) transformations. We are in the process of migrating from psyGen to psyir/transformations so there are actually two places at the moment. This obviously makes things more complicated rather than helping but I thought it was worth mentioning.
